prototypes/ItemRepeaterPrototypes/ItemsRepeaterExperiments/AttachedBehaviors/SelectionModelPage.xaml.cs

```diff
@@ -0,0 +1,101 @@
+﻿using ItemsRepeaterExperiments.Common;
+using Microsoft.UI.Xaml.Controls;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Runtime.InteropServices.WindowsRuntime;
+using Windows.Foundation;
+using Windows.Foundation.Collections;
+using Windows.UI.Xaml;
+using Windows.UI.Xaml.Controls;
+using Windows.UI.Xaml.Controls.Primitives;
+using Windows.UI.Xaml.Data;
+using Windows.UI.Xaml.Input;
+using Windows.UI.Xaml.Media;
+using Windows.UI.Xaml.Navigation;
+
+// The Blank Page item template is documented at https://go.microsoft.com/fwlink/?LinkId=234238
+
+namespace ItemsRepeaterExperiments.AttachedBehaviors
+{
+    /// <summary>
+    /// An empty page that can be used on its own or navigated to within a Frame.
+    /// </summary>
+    public sealed partial class SelectionModelPage : Page
+    {
+        private List<Team> teams = DataSourceCreator<Team>.CreateRandomizedList(100);
+        public SelectionModelPage()
+        {
+            this.InitializeComponent();
+        }
+
+
+        /// <summary>
+        /// Attach properties to virtualized items and update containers
+        /// </summary>
+        /// <param name="sender"></param>
+        /// <param name="args"></param>
+        private void ItemsRepeater_ElementPrepared(ItemsRepeater sender, ItemsRepeaterElementPreparedEventArgs args)
+        {
+            var control = args.Element as FrameworkElement;
+
+            // Attach repeater and selectionmodel properties for handling of selection
+            SelectionBehavior.SetParentItemsRepeater(control, TeamPresenter);
+            SelectionBehavior.SetSelectionModel(control, selectionModel);
+
+            SelectionBehavior.SetIsSelected(control,
+                selectionModel.SelectedItems.Contains(teams[TeamPresenter.GetElementIndex(control)]));
+        }
```

Comment:
These could be done if SelectionBehavior was attached on ItemsRepeater :) . If SelectionBehavior is attached on ItemsRepeater, perhaps we don't have to store it on each element. Each element could have the parent ItemsRepeater and we can get to the SelectionModel through that. With that change, we would have effectively moved all the wire up outside the control and into the behaviors. 

This would mean SelectionBehavior knows about SelectionModel and ItemsRepeater, but it does make the usage a lot simpler)

Reply:
I've thought about that too. So `IsEnabled` is only something being applied to the ItemsRepeater, while `IsSelected` is something we apply on the realized items.

I'll try to get that working now.

Reply:
Maybe we need to have IsEnabled be both applied to the ItemsRepeater and the realized children so we can register events to all of them and unregister them as needed.

Reply:
If we can attach IsEnabled to ItemsRepeater, then we could register/unregister on the children during ItemsRepeater.ElementPrepared and unregister in ItemsRepeater.ElementClearing. Getting all the realized elements seems to come up over and over, I think we should add that API.

Reply:
Updated the behavior now to so we only have selectionchanged outside the behavior.